sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -991,6 +999,16 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
             Row(null, null, null, "]") :: Nil
         )
 
+        // Check if corrupt records are dropped.
```

Comment:
No, it drops because it selects the table `jsonTableWithDropMalformed`. I did not add a test for `PERMISSIVE` mode because JSON data source has been `PERMISSIVE` mode itself.